docs/tests.md

```diff
@@ -1,27 +1,242 @@
+the base tests.
+
+> *End of test group discussion*
+
+### Test types
+
+There are three PURL test types defined in the current v0.1 test schema:
+
+- **build**: "A PURL building test from decoded components to a canonical PURL
+ string.".
+- **parse**: "A PURL parsing test from a canonical PURL string to decoded 
+components.".
+- **roundtrip**: "A PURL roundtrip test, parsing then building back a PURL 
+from a canonical string input."
+
+>*Discussion and proposed changes for test types*
+
+The current definition for the **roundtrip** test type does not make sense 
+because it requires a canonical string input and output. The proposed new 
+schema definitions for test types are:
+
+- **build**: A test to build a canonical PURL output string from an input of 
+decoded PURL components."
+- **parse**: A test to parse decoded components from a PURL input string."
+- **roundtrip**: A test to produce a canonical PURL output string from an 
+input PURL string.
+
+For the **build** and **parse** test types there is no normative change here -
+ just an attempt to improve the text.
+
+#### New validation test type
+
+There is a proposed new **validation** test type from [Issue 692](https://github.com/package-url/purl-spec/issues/692) and [PR 614](https://github.com/package-url/purl-spec/pull/614). 
+The proposed definition is: "A PURL validation test, checking if a PURL 
+follows all the rules for a particular ecosystem."
+- A better definition (editorially) would be: "A test to validate that an 
+input PURL string complies with the rules for its PURL type." 
+- The new PURL type test cases in PR 614 are a mixture of **base** and
+**advanced** test groups, but there is no obvious pattern for the difference.
+
+#### Validation test messages:
+
+PR 614 adds a **purl_validation_message** to the test schema with possible 
+validation severity values of "info", "warning" or "error". There is no 
+specific definition of the difference between info and warning messages, but 
+the examples in PURL type test case updates from PR 614 follow the patterns:
+- Warning messages for cases where a PURL component does not follow rules 
+related to case sensitivity.
+- Info messages for cases of an unregistered namespace or qualifier - see also
+Issues [690](https://github.com/package-url/purl-spec/issues/690) and 
+[710](https://github.com/package-url/purl-spec/issues/710).
+
+Validation message notes:
+- The proposed change for validation messages allows multiple messages from a 
+single test case.
+- There is no message returned for a successful validation.
+- We will need to clearly define the difference between an info message and a 
+warning message or reduce the set to warning and error messages.
+- The messages as shown in the test case examples from PR 614 are clear and 
+helpful as presented, but it is not clear if/how some PURL tools will be able 
+to handle such specific messages. See also the following Error 
```

Comment:
We are not expecting tools to report the exact same message, just to report a problem of the same type IMHO